server/src/main/java/org/elasticsearch/action/search/SearchShardIterator.java

```diff
@@ -54,6 +54,28 @@ public SearchShardIterator(@Nullable String clusterAlias, ShardId shardId, List<
 
     /**
      * Creates a {@link SearchShardIterator} instance that iterates over a subset of the given shards
+     * for a given <code>shardId</code>.
+     *
+     * @param clusterAlias    the alias of the cluster where the shard is located
+     * @param shardId         shard id of the group
+     * @param shards          shards to iterate
+     * @param originalIndices the indices that the search request originally related to (before any rewriting happened)
+     * @param skip            if true, then this group won't have matches (due to an index level block),
+     *                        and it can be safely skipped from the search
+     */
+    public SearchShardIterator(
```

Comment:
Small additional nit: this constructor seems to replace the older with the (clusterAlias, shardId, shards, originalIndices) signature. The older constructor only seems to be used in test code after that. If that's really the case, I think we should remove the older ctor and replace calls in test code with the newer version and `skip==false`. I think its always good to keep the entry points for a class as small as possible.